### PR TITLE
imagemagick: Depends on freetype [WIP; DNM]

### PR DESCRIPTION
**Pull request by aghasemi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

 It seems to need it for being built. See issue https://github.com/Linuxbrew/brew/issues/170.

### CODE REVIEW

File: Formula/imagemagick.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ class Imagemagick < Formula
   deprecated_option "with-jp2" => "with-openjpeg"
 
   depends_on "pkg-config" => :build
+  depends_on "freetype" => :build
```
Comment (sjackman) [inline Formula/imagemagick.rb]: Please change this line to
```ruby
depends_on "freetype" unless OS.mac?
```
and move it to line 40 right after `xz`.

**Comment by sjackman** [comment]
Please change the git commit log message to
`imagemagick: Depends on freetype for Linuxbrew`

**Comment by sjackman** [comment]
Hmm. This issue appears more complicated than I first thought. This formula already has a `:recommended` dependency on `freetype`.
```ruby
depends_on "freetype" => :recommended
```
https://github.com/aghasemi/homebrew-core/blob/5eb39f4e9d19a9a55bb6a15bfac08bb05cf0d74d/Formula/imagemagick.rb#L44

That should have cause `freetype` to be installed on your system before `imagemagick`.

**Comment by sjackman** [comment]
https://gist.github.com/anonymous/3f045200226d67712019084745730e3f#file-02-make-L23
`libtool: compile:  /usr/local/linuxbrew/bin/gcc-4.8 -std=gnu99 -std=gnu99 -DHAVE_CONFIG_H -I. -I./config -I. -I. -isystem/usr/local/linuxbrew/include -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I@@HOMEBREW_CELLAR@@/libpng/1.6.25/include/libpng16 -I@@HOMEBREW_CELLAR@@/xz/5.2.2/include -I@@HOMEBREW_PREFIX@@/opt/freetype/include/freetype2 -I@@HOMEBREW_CELLAR@@/zlib/1.2.8/include -I@@HOMEBREW_CELLAR@@/libpng/1.6.25/include/libpng16 -I@@HOMEBREW_CELLAR@@/zlib/1.2.8/include -Os -w -pipe -march=native -Wall -mtune=corei7-avx -fexceptions -pthread -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -c coders/tga.c -o coders/coders_tga_la-tga.o >/dev/null 2>&1`

For some strange reason `@@HOMEBREW_CELLAR@@` isn't properly being expanded.

**Comment by sjackman** [comment]
As a workaround, if you're able to install Linuxbrew in `/home/linuxbrew/.linuxbrew` you'll be able to use the precompiled binary bottle of `imagemagick`.

**Comment by sjackman** [comment]
Did this PR resolve your issue? Based on your `gist` log, it doesn't appear that it should.

**Comment by aghasemi** [comment]
No. Not solved yet. Freetype is already installed. I get this
```
Warning: freetype-2.7 already installed
==> Installing dependencies for imagemagick: pkg-config, libtool, bzip2, jpeg, libtiff
==> Installing imagemagick dependency: pkg-config
==> Downloading https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.1.tar.gz
Already downloaded: /home/chronos/user/.cache/Homebrew/pkg-config-0.29.1.tar.gz
==> ./configure --prefix=/usr/local/linuxbrew/Cellar/pkg-config/0.29.1_2 --disable-host-tool --with-internal-glib --with-pc-path=/usr/local/linuxbrew/lib/pkgconfig:/usr
Last 15 lines from /home/chronos/user/.cache/Homebrew/Logs/pkg-config/01.configure:
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... /usr/local/linuxbrew/bin/gcc-4.8
checking whether the C compiler works... no
configure: error: in `/usr/local/linuxbrew/tmp/pkg-config-20161105-31552-dzws51/pkg-config-0.29.1':
configure: error: C compiler cannot create executables
See `config.log' for more details

READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/Linuxbrew/homebrew-core/issues
```

**Comment by aghasemi** [comment]
https://gist.github.com/84719f68b8830b2669999f7b9313eaa1

**Comment by aghasemi** [comment]
It was because I had unset LD_LIBRARY_PAT. Now it compiles.

**Comment by sjackman** [comment]
Setting `LD_LIBRARY_PATH` may break other things, but I'm glad to hear that you got it working.

**Pull request closed without merging**